### DISCUSSION TITLE
Настройка отправки email

### DISCUSSION THREAD

**Discussion by fzaur** [totumonline/totum-ru-issues-and-discussions#136] [Как это сделать?] [answered]
Добрый день. Не подскажите как можно настроить отправку email. В ходе начальной установки я пропустил этот этап, потому как мы используем почту на яндексе (днс делегирован на яндексе и почта соответственно с нашим доменом на нем). Инструкция в доках показывает как это сделать, но ведь у нас меил-сервером является другая служба. Как быть в этом случае?

**Comment by Qb1cc** [comment, marked as answer]
Попробуй загугли, настройка EXIM для Yandex , там есть куча гайдов

**Reply by ttm-alexeypolunin** [reply]
Да, это вариант. Вместо настройки exim как в инструкции настроить его на релей к учетке на яндексе.

Еще варинт настроить Conf тотума, что бы он не sendmail использовал а SMTP — я чуть позже инструкцию сделаю.

**Reply by ttm-alexeypolunin** [reply]
## Как перенастроить базовый `PhpMailer` с `sendmail` на `SMTP` в `Conf.php` для всех схем:

Открываем на редактирование Conf.php в корне установки:

1. Меняем `use totum\common\configs\WithPhpMailerTrait;` на `use totum\common\configs\WithPhpMailerSmtpTrait;`
2. Меняем `use WithPhpMailerTrait;` на `use WithPhpMailerSmtpTrait;`
3. Добавляем со следующей строки параметры SMTP-подключения:
```
protected $SmtpData = [
                'host' => 'YOU_HOST_HERE',
                'port' => 25,
                'login' => 'YOU_LOGIN_HERE',
                'pass' => 'YOU_PASS_HERE',
            ];
```

**Reply by ttm-alexeypolunin** [reply]
## Настройка SMTP через настройки схемы

С версии 3.6.45 добавлены параметры кастомного адреса отправки по умолчанию и кастомные настройки SMTP:

<img width="827" alt="Снимок экрана 2022-08-03 в 14 49 04" src="https://user-images.githubusercontent.com/55755565/182600962-d00f38eb-760b-4fd6-b1a8-75c0b44f3a6c.png">

**Если настроены `custom_smtp_setings_for_schema` то трейт и прочие настройки  в `Conf.php` не нужны!**

**Если настроен трейт в Conf.php настройка `custom_smtp_setings_for_schema` не подтягивается.**

Заполняется как массив:

```
{
  "host": "ssl://smtp.gmail.com",
  "port": 465,
  "login": "totum@totum.online",
  "password": "password_here"
}
```

Обязательно смотрите инструкцию к своему SMTP — обычно есть еще аппрув со стороны SMTP-сервера.

**Comment by poluninatm** [comment]
## Кастомный способ отправки email

Если нужно совсем как-то необычно извратиться (например, запустить отправку писем через шлюз с неким API) и есть скилл в PHP, можно убрать `use WithPhpMailerTrait;` в `Conf.php` и добавить там функцию `public function sendMail` — по образцу трейта `WithPhpMailerTrait`. 

Собственно и `WithPhpMailerTrait`, и `WithPhpMailerSmtpTrait` определяют эту функцию для отправки через шлюзы `sendmail` и `SMTP` с помощью `PhpMailer`.
